### PR TITLE
core: use proxy for cover/poster images url. resolves #12761

### LINKED ISSUE #12761: [req]: Always serve requests directly from Jackett and proxy all requests
### Is there already a request for your feature?

- [X] I have checked older issues, open and closed

### Is your feature request related to a problem? Please describe.

Images shown during mouseover are not proxied.

To replicate:
1. Open Jackett.
2. Enable the logger in uBlock Origin.
3. Search JPopsuki for any torrent.
4. Hover your mouse over the name of the torrent.
5. Observe there is a request to https://jpopsuki.eu/static/images/torrents/*.th.jpg.

### Describe the solution you'd like

Always serve requests directly from Jackett. Proxy all requests (including images shown during mouseover) if the proxy option is enabled.

### Describe alternatives you've considered

Allow users to disable image previews on mouseover.

### PR DESCRIPTION
**Pull request by ngosang**

### CODE REVIEW

**Comment by ngosang** [comment]
@ilike2burnthing I think its OK. Just test it a bit before merging.
You should see the URL of all images/covers/posters are Jackett URLs. From now we treat the images the same as the torrent download. We send the Cookies and Referrer header to request images. This could fix some issues related to not working posters in Jackett UI.

**Comment by ilike2burnthing** [comment]
Working well, with and without proxy

File: src/Jackett.Server/Controllers/IndexerApiController.cs [hunk 1]
```diff
@@ -167,11 +167,11 @@ private void ConfigureCacheResults(IEnumerable<TrackerCacheResult> results)
...
             {
-                var link = result.Link;
                 var file = StringUtil.MakeValidFileName(result.Title, '_', false);
-                result.Link = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "dl", file);
+                result.Link = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "dl", file);
+                result.Poster = serverService.ConvertToProxyLink(result.Poster, serverUrl, result.TrackerId, "img", "poster");
                 if (result.Link != null && result.Link.Scheme != "magnet" && !string.IsNullOrWhiteSpace(serverService.GetBlackholeDirectory()))
-                    result.BlackholeLink = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "bh", file);
+                    result.BlackholeLink = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "bh", file);
             }
```
Comment (garfield69) [inline src/Jackett.Server/Controllers/IndexerApiController.cs:175]: looks like result.BlackholeLink is getting result.Link converted twice by serverService.ConvertToProxyLink, once at L171 and again at L174

File: src/Jackett.Server/Controllers/ResultsController.cs [hunk 2]
```diff
@@ -531,13 +535,13 @@ private void ConfigureCacheResults(IEnumerable<TrackerCacheResult> results)
...
                 var file = StringUtil.MakeValidFileName(result.Title, '_', false);
-                result.Link = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "dl", file);
+                result.Link = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "dl", file);
+                result.Poster = serverService.ConvertToProxyLink(result.Poster, serverUrl, result.TrackerId, "img", "poster");
                 if (!string.IsNullOrWhiteSpace(serverConfig.BlackholeDir))
                 {
                     if (result.Link != null)
-                        result.BlackholeLink = serverService.ConvertToProxyLink(link, serverUrl, result.TrackerId, "bh", file);
+                        result.BlackholeLink = serverService.ConvertToProxyLink(result.Link, serverUrl, result.TrackerId, "bh", file);
                     else if (result.MagnetUri != null)
```
Comment (garfield69) [inline src/Jackett.Server/Controllers/ResultsController.cs:545]: and same here

**Pull request merged**